src/chat.rs

```diff
@@ -3314,20 +3315,61 @@ pub async fn marknoticed_chat(context: &Context, chat_id: ChatId) -> Result<()>
             context.emit_event(EventType::MsgsNoticed(chat_id_in_archive));
             chatlist_events::emit_chatlist_item_changed(context, chat_id_in_archive);
         }
-    } else if context
-        .sql
-        .execute(
-            "UPDATE msgs
-            SET state=?
-          WHERE state=?
-            AND hidden=0
-            AND chat_id=?;",
-            (MessageState::InNoticed, MessageState::InFresh, chat_id),
-        )
-        .await?
-        == 0
-    {
-        return Ok(());
+    } else {
+        let conn_fn = |conn: &mut rusqlite::Connection| {
+            // This is to trigger emitting `MsgsNoticed` on other devices when reactions are noticed
+            // locally. We filter out `InNoticed` messages because they are normally a result of
+            // `mark_old_messages_as_noticed()` which happens on all devices anyway. Also we limit
+            // this to one message because the effect is the same anyway.
```

Comment:
I would have assumed the previous limit of 100 to be more efficient, because all reactions need to be marked as seen eventually, and it's more efficient to mark all of them at once (although I do like having a limit at all, because otherwise the UI may become unresponsive when too many messages are marked as seen at once)

Reply:
We don't need to mark all reactions as seen on IMAP, we only want to trigger `MsgsNoticed` on other devices, so we only select the last hidden message (reaction) and check if it's `InFresh`. Anyway we don't send MDNs for reactions. If a reactions is already `InNoticed` or `InSeen`, it normally has the same state on other devices, so there's nothing to do, that's why i added `filter()` below.

Reply:
No, i added `LIMIT 1`, but i select `InFresh`, `InNoticed` and `InSeen` hidden messages (reactions) and then check that the selected reaction is `InFresh` (the `filter()` call below).

For uniformity it would be better to mark all reactions as seen eventually, but it's an extra work on IMAP that i suggest to avoid at least for now.